features/305-microservice/microservice.md

```diff
@@ -14,16 +16,48 @@ deployments:
     port: 8888
     context: ./backend/service
     dockerfile: Dockerfile
+    command: ["<start server command in array>"]
 ```
 
-When the user run `skycli app deploy my-awesome-service`, the following steps are taken:
+*`dockerfile`, `command` are optional.*
+
+When the user run `skycli app deploy`, the following steps are taken:
 
 1. Assert `./backend/service` to be a directory.
 1. If `dockerfile` is specified, it is relative to `context`. Otherwise, the default value is `context/Dockerfile`.
 1. Assert the Dockerfile to be a file.
 1. Look for `<context>/.dockerignore`. If it is not present, the entire context is archived. Otherwise, use the same mechanism as `.dockerignore` to ignore files.
 1. Upload the archive and save as an artifact.
-1. Follow the existing deployment flow.
+1. Follow the existing docker deployment flow.
+1. `command` is optional. If it is provided, it will be set as `args` of container spec in k8s.
+
+
+### Microservice with preconfigured environment (e.g. nodejs)
+
+```yaml
+deployments:
+  my-awesome-service:
+    type: http-service
+    path: /api/
+    port: 3000
```

Comment:
I'm a bit confused if the docker template is provided by us when the template is present, what is the point of specifying port?

Reply:
User can start the web server at any port which is specified in their code, so we need user to provide the port number in skygear.yaml

Reply:
ah ic, sorry for my stupidity